x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ssl/SSLService.java

```diff
@@ -430,27 +425,47 @@ private SSLContextHolder createSslContext(X509ExtendedKeyManager keyManager, X50
      * Parses the settings to load all SSLConfiguration objects that will be used.
      */
     Map<SSLConfiguration, SSLContextHolder> loadSSLConfigurations() {
-        Map<SSLConfiguration, SSLContextHolder> sslConfigurations = new HashMap<>();
-        sslConfigurations.put(globalSSLConfiguration, createSslContext(globalSSLConfiguration));
+        Map<SSLConfiguration, SSLContextHolder> sslContextHolders = new HashMap<>();
+        sslContextHolders.put(globalSSLConfiguration, createSslContext(globalSSLConfiguration));
+        this.sslConfigurations.put("_global", globalSSLConfiguration);
 
         final Settings transportSSLSettings = settings.getByPrefix(XPackSettings.TRANSPORT_SSL_PREFIX);
-        List<Settings> sslSettingsList = new ArrayList<>();
-        sslSettingsList.add(getHttpTransportSSLSettings(settings));
-        sslSettingsList.add(settings.getByPrefix("xpack.http.ssl."));
-        sslSettingsList.addAll(getRealmsSSLSettings(settings));
-        sslSettingsList.addAll(getMonitoringExporterSettings(settings));
-
-        sslSettingsList.forEach((sslSettings) ->
-                sslConfigurations.computeIfAbsent(new SSLConfiguration(sslSettings, globalSSLConfiguration), this::createSslContext));
+        Map<String, Settings> sslSettingsMap = new HashMap<>();
+        sslSettingsMap.put(XPackSettings.HTTP_SSL_PREFIX, getHttpTransportSSLSettings(settings));
+        sslSettingsMap.put("xpack.http.ssl", settings.getByPrefix("xpack.http.ssl."));
+        sslSettingsMap.putAll(getRealmsSSLSettings(settings));
+        sslSettingsMap.putAll(getMonitoringExporterSettings(settings));
+
+        sslSettingsMap.forEach((key, sslSettings) -> {
+            if (sslSettings.isEmpty()) {
+                storeSslConfiguration(key, globalSSLConfiguration);
+            } else {
+                final SSLConfiguration configuration = new SSLConfiguration(sslSettings, globalSSLConfiguration);
+                storeSslConfiguration(key, configuration);
+                sslContextHolders.computeIfAbsent(configuration, this::createSslContext);
+            }
+        });
 
         // transport is special because we want to use a auto-generated key when there isn't one
         final SSLConfiguration transportSSLConfiguration = new SSLConfiguration(transportSSLSettings, globalSSLConfiguration);
         this.transportSSLConfiguration.set(transportSSLConfiguration);
-        List<Settings> profileSettings = getTransportProfileSSLSettings(settings);
-        sslConfigurations.computeIfAbsent(transportSSLConfiguration, this::createSslContext);
-        profileSettings.forEach((profileSetting) ->
-            sslConfigurations.computeIfAbsent(new SSLConfiguration(profileSetting, transportSSLConfiguration), this::createSslContext));
-        return Collections.unmodifiableMap(sslConfigurations);
+        this.sslConfigurations.put("_transport", transportSSLConfiguration);
```

Comment:
the use of `_` as our prefix to avoid collisions or to make something special concerns me. In the IpFilter we use `.http` for HTTP to avoid collisions since it is much harder to create a setting/name that would result in that as the key. Maybe we should use `.` instead of `_` here?

Reply:
Actually I have no idea _why_ I'm naming this one in a special way. It should just be `xpack.security.transport.ssl` the only special one should be `.global`

Reply:
Scrap that, `.global` should just be `xpack.ssl`